src/python/pants/util/memo.py

```diff
@@ -24,6 +26,44 @@ def equal_args(*args, **kwargs):
   return key
 
 
+def coercing_arg_normalizer(type_coercions, base_normalizer, args, kwargs):
```

Comment:
It would be wonderful to keep consistent terminology in a single file. base_normalizer is elsewhere key_factory. Maybe `coercing_key_factory` and `base_key_factory` or `underlying` or else rename everything related in this file to *normalizer.